File: EncounterSimulator/Models/Character.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Data.SqlClient;
using LyeltDatabaseConnector;

namespace EncounterSimulator.Models
{
    public class AvailableCharacter
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public int MaxHP { get; set; }

        public int AC { get; set; }

        public int Speed { get; set; }

        public string Owner { get; set; }

        public int DexModifier { get; set; }

        public AvailableCharacter(int id, string name, int maxHP, int ac, int speed, string owner, int dexMod)
        {
            Id = id;
            Name = name;
            MaxHP = maxHP;
            AC = ac;
            Speed = speed;
            Owner = owner;
            DexModifier = dexMod;
        }

        public AvailableCharacter() { }
    }

    public class ActiveCharacter : AvailableCharacter
    {
        public List<Status> Statuses { get; set; }

        public int CurrentHP { get; set; }
    }
}

File: EncounterSimulator/src/models/character.ts
export class AvailableCharacter {
    id: number;

    name: string;

    maxHP: number;

    ac: number;

    speed: number;

    owner: string;

    dexModifier: number;

    initiativeRoll: number;

    constructor(character?: AvailableCharacter) {
        if (character && character != null) {
            this.id = character.id;
            this.name = character.name;
            this.maxHP = character.maxHP;
            this.ac = character.ac;
            this.speed = character.speed;
            this.owner = character.owner;
            this.dexModifier = character.dexModifier;
            this.initiativeRoll = character.initiativeRoll;
        }
    }
}

export class ActiveCharacter extends AvailableCharacter {
    statuses: Status[];

    currentHP: number;

    constructor(original: AvailableCharacter) {
        super(original);
        this.statuses = [];
        this.currentHP = this.maxHP;
    }
}

export class Status {
    id: number;

    name: string;

    description: string;

    duration: number;
}

export class Action {
    id: number;

    name: string;

    description: string;
}